examples/typescript/passkey/src/recoverPubKey.ts

```diff
@@ -0,0 +1,65 @@
+import EC from "elliptic";
+const ec = new EC.ec("secp256r1");
+
+export async function sha256(message: Uint8Array) {
+  const hashBuffer = await crypto.subtle.digest("SHA-256", message);
+  return new Uint8Array(hashBuffer);
+}
+
+export function concatenateUint8Arrays(array1: Uint8Array, array2: Uint8Array) {
+  const result = new Uint8Array(array1.length + array2.length);
+  result.set(array1, 0);
+  result.set(array2, array1.length);
+
+  return result;
+}
+
+export async function recoverPublicKey(
+  authenticatorData: Uint8Array,
+  clientDataJSON: Uint8Array,
+  signature: Uint8Array,
+) {
+  const shaClientDataJSON = await sha256(clientDataJSON);
+  const message = concatenateUint8Arrays(authenticatorData, shaClientDataJSON);
+
+  const result = await recoverPublicKeyFromMessageAndSignature(message, signature);
+  return result;
+}
+
+export async function recoverPublicKeyFromMessageAndSignature(message: Uint8Array, signature: Uint8Array) {
```

Comment:
Thanks for adding this in. 

1. Can we add tests for these methods that use the Passkey classes below? We want to make sure users can recover the public key associated with their passkey credential from one or more valid ECDSA signatures in the event that that the relying party loses or misplaces the user's public key. 
2. Also lets include a button in the UI for the `examples/typescript/passkey` folder that tests this